### ISSUE TITLE
"chisel dev" should kill itself if the forked "chiseld" terminates

### ISSUE THREAD

**Issue by penberg** [chiselstrike/chiselstrike#540]

**Comment by penberg** [comment]
@honzasp We had @haaawk test this and the issue seems to have been fixed. Do you know if your big refactoring pull request could have fixed this?

**Comment by haaawk** [comment]
Experiment report:
```
% git diff
diff --git a/server/src/http.rs b/server/src/http.rs
index e1456af3..f2fdbedd 100644
--- a/server/src/http.rs
+++ b/server/src/http.rs
@@ -136,6 +136,8 @@ async fn handle_version_request(
     let (req_parts, req_body) = request.into_parts();
     let req_body = hyper::body::to_bytes(req_body).await?;
 
+    panic!("haaawk");
+
     // TODO: we don't authenticate the user!!!
     let user_id: Option<String> = req_parts
         .headers

% ps aux | grep chisel
haaawk            6120   0.1  0.2 40813800 102752 s003  S+   12:00PM   0:01.07 /chiselstrike/my-backend/../target/debug/chiseld --debug
haaawk            6132   0.0  0.0 33589096    312 s000  R+   12:00PM   0:00.00 grep chisel
haaawk            6119   0.0  0.0 34363296  13364 s003  S+   12:00PM   0:00.04 ../target/debug/chisel dev

% curl localhost:8080/dev/hello
curl: (52) Empty reply from server

% ps aux | grep chisel
haaawk            6135   0.0  0.0 34253900    948 s000  S+   12:00PM   0:00.00 grep chisel
```

**Comment by haaawk** [comment]
It seems that we're waiting for forked process:
https://github.com/chiselstrike/chiselstrike/blob/main/cli/src/main.rs#L163-L170

**Comment by penberg** [comment]
@MarinPostma Did you fix this in 96e009340661f84794cc5b199898d091820772d4?

**Comment by MarinPostma** [comment]
long time ago, but yes, looks like this was the intended effect
